### ISSUE TITLE
Issue when generating dataset with scale=2

### ISSUE THREAD

**Issue by yestinchen** [uwdb/visualroad#7]
Hi,

Thanks for the great work!

Recently I've been working on processing specific queries on videos and I would like to use your generator as a benchmark. However, when I try to generator the dataset with `scale=2`, it stopped during generation and gave me an error.

Could you please help me out? Am I missing something? Many thanks!

The command I am using is `sudo docker-compose run generator --scale 2 dataset-test-2`

The output is:
```
4.22.0-0+++UE4+Release-4.22 517 0
Disabling core dumps.
INFO:root:Map: Town03, Weather: WeatherParameters(cloudyness=80.000000, precipitation=60.000000, precipitation_deposits=100.000000, wind_intensity=1.000000, sun_azimuth_angle=0.000000, sun_altitude_angle=15.000000), Vehicles: 200, Walkers: 250
INFO:root:Tile 1 of 2: Rendered 0 frames; 900 remaining (0.0 FPS)
INFO:root:Tile 1 of 2: Rendered 0 frames; 900 remaining (0.0 FPS)
INFO:root:Tile 1 of 2: Rendered 0 frames; 900 remaining (0.0 FPS)
INFO:root:Tile 1 of 2: Rendered 0 frames; 900 remaining (0.0 FPS)
INFO:root:Tile 1 of 2: Rendered 0 frames; 900 remaining (0.0 FPS)
INFO:root:Tile 1 of 2: Rendered 0 frames; 900 remaining (0.0 FPS)
INFO:root:Tile 1 of 2: Rendered 0 frames; 900 remaining (0.0 FPS)
INFO:root:Tile 1 of 2: Rendered 0 frames; 900 remaining (0.0 FPS)
INFO:root:Tile 1 of 2: Rendered 0 frames; 900 remaining (0.0 FPS)
INFO:root:Tile 1 of 2: Rendered 0 frames; 900 remaining (0.0 FPS)
INFO:root:Tile 1 of 2: Rendered 8 frames; 892 remaining (0.2 FPS)
INFO:root:Tile 1 of 2: Rendered 18 frames; 882 remaining (0.3 FPS)
INFO:root:Tile 1 of 2: Rendered 28 frames; 872 remaining (0.5 FPS)
INFO:root:Tile 1 of 2: Rendered 38 frames; 862 remaining (0.6 FPS)
INFO:root:Tile 1 of 2: Rendered 48 frames; 852 remaining (0.7 FPS)
INFO:root:Tile 1 of 2: Rendered 58 frames; 842 remaining (0.8 FPS)
INFO:root:Tile 1 of 2: Rendered 68 frames; 832 remaining (0.9 FPS)
INFO:root:Tile 1 of 2: Rendered 78 frames; 822 remaining (1.0 FPS)
INFO:root:Tile 1 of 2: Rendered 88 frames; 812 remaining (1.0 FPS)
INFO:root:Tile 1 of 2: Rendered 98 frames; 802 remaining (1.1 FPS)
INFO:root:Tile 1 of 2: Rendered 108 frames; 792 remaining (1.1 FPS)
INFO:root:Tile 1 of 2: Rendered 118 frames; 782 remaining (1.2 FPS)
INFO:root:Tile 1 of 2: Rendered 128 frames; 772 remaining (1.2 FPS)
INFO:root:Tile 1 of 2: Rendered 138 frames; 762 remaining (1.3 FPS)
INFO:root:Tile 1 of 2: Rendered 148 frames; 752 remaining (1.3 FPS)
INFO:root:Tile 1 of 2: Rendered 158 frames; 742 remaining (1.4 FPS)
INFO:root:Tile 1 of 2: Rendered 168 frames; 732 remaining (1.4 FPS)
INFO:root:Tile 1 of 2: Rendered 178 frames; 722 remaining (1.4 FPS)
INFO:root:Tile 1 of 2: Rendered 188 frames; 712 remaining (1.4 FPS)
INFO:root:Tile 1 of 2: Rendered 198 frames; 702 remaining (1.5 FPS)
INFO:root:Tile 1 of 2: Rendered 208 frames; 692 remaining (1.5 FPS)
INFO:root:Tile 1 of 2: Rendered 218 frames; 682 remaining (1.5 FPS)
INFO:root:Tile 1 of 2: Rendered 228 frames; 672 remaining (1.5 FPS)
INFO:root:Tile 1 of 2: Rendered 238 frames; 662 remaining (1.6 FPS)
INFO:root:Tile 1 of 2: Rendered 248 frames; 652 remaining (1.6 FPS)
INFO:root:Tile 1 of 2: Rendered 258 frames; 642 remaining (1.6 FPS)
INFO:root:Tile 1 of 2: Rendered 268 frames; 632 remaining (1.6 FPS)
INFO:root:Tile 1 of 2: Rendered 278 frames; 622 remaining (1.6 FPS)
INFO:root:Tile 1 of 2: Rendered 288 frames; 612 remaining (1.7 FPS)
INFO:root:Tile 1 of 2: Rendered 298 frames; 602 remaining (1.7 FPS)
INFO:root:Tile 1 of 2: Rendered 308 frames; 592 remaining (1.7 FPS)
INFO:root:Tile 1 of 2: Rendered 318 frames; 582 remaining (1.7 FPS)
INFO:root:Tile 1 of 2: Rendered 328 frames; 572 remaining (1.7 FPS)
INFO:root:Tile 1 of 2: Rendered 338 frames; 562 remaining (1.7 FPS)
INFO:root:Tile 1 of 2: Rendered 348 frames; 552 remaining (1.7 FPS)
INFO:root:Tile 1 of 2: Rendered 358 frames; 542 remaining (1.7 FPS)
INFO:root:Tile 1 of 2: Rendered 368 frames; 532 remaining (1.8 FPS)
INFO:root:Tile 1 of 2: Rendered 378 frames; 522 remaining (1.8 FPS)
INFO:root:Tile 1 of 2: Rendered 388 frames; 512 remaining (1.8 FPS)
INFO:root:Tile 1 of 2: Rendered 398 frames; 502 remaining (1.8 FPS)
INFO:root:Tile 1 of 2: Rendered 408 frames; 492 remaining (1.8 FPS)
INFO:root:Tile 1 of 2: Rendered 418 frames; 482 remaining (1.8 FPS)
INFO:root:Tile 1 of 2: Rendered 428 frames; 472 remaining (1.8 FPS)
INFO:root:Tile 1 of 2: Rendered 438 frames; 462 remaining (1.8 FPS)
INFO:root:Tile 1 of 2: Rendered 448 frames; 452 remaining (1.8 FPS)
INFO:root:Tile 1 of 2: Rendered 458 frames; 442 remaining (1.8 FPS)
INFO:root:Tile 1 of 2: Rendered 468 frames; 432 remaining (1.8 FPS)
INFO:root:Tile 1 of 2: Rendered 478 frames; 422 remaining (1.9 FPS)
INFO:root:Tile 1 of 2: Rendered 488 frames; 412 remaining (1.9 FPS)
INFO:root:Tile 1 of 2: Rendered 498 frames; 402 remaining (1.9 FPS)
INFO:root:Tile 1 of 2: Rendered 508 frames; 392 remaining (1.9 FPS)
INFO:root:Tile 1 of 2: Rendered 518 frames; 382 remaining (1.9 FPS)
INFO:root:Tile 1 of 2: Rendered 528 frames; 372 remaining (1.9 FPS)
INFO:root:Tile 1 of 2: Rendered 538 frames; 362 remaining (1.9 FPS)
INFO:root:Tile 1 of 2: Rendered 548 frames; 352 remaining (1.9 FPS)
INFO:root:Tile 1 of 2: Rendered 558 frames; 342 remaining (1.9 FPS)
INFO:root:Tile 1 of 2: Rendered 568 frames; 332 remaining (1.9 FPS)
INFO:root:Tile 1 of 2: Rendered 578 frames; 322 remaining (1.9 FPS)
INFO:root:Tile 1 of 2: Rendered 588 frames; 312 remaining (1.9 FPS)
INFO:root:Tile 1 of 2: Rendered 598 frames; 302 remaining (1.9 FPS)
INFO:root:Tile 1 of 2: Rendered 608 frames; 292 remaining (1.9 FPS)
INFO:root:Tile 1 of 2: Rendered 618 frames; 282 remaining (1.9 FPS)
INFO:root:Tile 1 of 2: Rendered 628 frames; 272 remaining (1.9 FPS)
INFO:root:Tile 1 of 2: Rendered 638 frames; 262 remaining (1.9 FPS)
INFO:root:Tile 1 of 2: Rendered 648 frames; 252 remaining (1.9 FPS)
INFO:root:Tile 1 of 2: Rendered 658 frames; 242 remaining (1.9 FPS)
INFO:root:Tile 1 of 2: Rendered 668 frames; 232 remaining (2.0 FPS)
INFO:root:Tile 1 of 2: Rendered 678 frames; 222 remaining (2.0 FPS)
INFO:root:Tile 1 of 2: Rendered 688 frames; 212 remaining (2.0 FPS)
INFO:root:Tile 1 of 2: Rendered 698 frames; 202 remaining (2.0 FPS)
INFO:root:Tile 1 of 2: Rendered 708 frames; 192 remaining (2.0 FPS)
INFO:root:Tile 1 of 2: Rendered 718 frames; 182 remaining (2.0 FPS)
INFO:root:Tile 1 of 2: Rendered 728 frames; 172 remaining (2.0 FPS)
INFO:root:Tile 1 of 2: Rendered 738 frames; 162 remaining (2.0 FPS)
INFO:root:Tile 1 of 2: Rendered 748 frames; 152 remaining (2.0 FPS)
INFO:root:Tile 1 of 2: Rendered 758 frames; 142 remaining (2.0 FPS)
INFO:root:Tile 1 of 2: Rendered 768 frames; 132 remaining (2.0 FPS)
INFO:root:Tile 1 of 2: Rendered 778 frames; 122 remaining (2.0 FPS)
INFO:root:Tile 1 of 2: Rendered 788 frames; 112 remaining (2.0 FPS)
INFO:root:Tile 1 of 2: Rendered 798 frames; 102 remaining (2.0 FPS)
INFO:root:Tile 1 of 2: Rendered 808 frames; 92 remaining (2.0 FPS)
INFO:root:Tile 1 of 2: Rendered 818 frames; 82 remaining (2.0 FPS)
INFO:root:Tile 1 of 2: Rendered 828 frames; 72 remaining (2.0 FPS)
INFO:root:Tile 1 of 2: Rendered 838 frames; 62 remaining (2.0 FPS)
INFO:root:Tile 1 of 2: Rendered 848 frames; 52 remaining (2.0 FPS)
INFO:root:Tile 1 of 2: Rendered 858 frames; 42 remaining (2.0 FPS)
INFO:root:Tile 1 of 2: Rendered 868 frames; 32 remaining (2.0 FPS)
INFO:root:Tile 1 of 2: Rendered 878 frames; 22 remaining (2.0 FPS)
INFO:root:Tile 1 of 2: Rendered 888 frames; 12 remaining (2.0 FPS)
INFO:root:Tile 1 of 2: Rendered 898 frames; 2 remaining (2.0 FPS)
INFO:root:Tile 1 of 2: Rendered 908 frames; -8 remaining (2.0 FPS)
INFO:root:Destroying actors
INFO:root:Generation complete for tile 0
INFO:root:Map: Town07, Weather: WeatherParameters(cloudyness=80.000000, precipitation=30.000000, precipitation_deposits=50.000000, wind_intensity=0.400000, sun_azimuth_angle=0.000000, sun_altitude_angle=75.000000), Vehicles: 100, Walkers: 400
Traceback (most recent call last):
  File "/app/generator.py", line 409, in <module>
    generate(args.path, tile_pool, args.scale, (args.width, args.height), args.duration, args.fov, args.seed)
  File "/app/generator.py", line 345, in generate
    generate_tile(client, path, id, used_tiles[-1], scale, resolution, duration, panorama_fov)
  File "/app/generator.py", line 232, in generate_tile
    client.load_world(tile.map)
RuntimeError: map not found
FUnixPlatformMisc::RequestExitWithStatus
FUnixPlatformMisc::RequestExit
```

The generated configuration file is 
```
duration: 30
hostname: localhost
name: dataset-test-2
panorama_fov: null
port: 2000
resolution:
  height: 540
  width: 960
scale: 2
seed: 0
tiles:
- cameras:
  - type: traffic
    videos:
    - /app/dataset-test-2/traffic-000.mp4
  - type: traffic
    videos:
    - /app/dataset-test-2/traffic-001.mp4
  - type: traffic
    videos:
    - /app/dataset-test-2/traffic-002.mp4
  - type: traffic
    videos:
    - /app/dataset-test-2/traffic-003.mp4
  - type: panoramic
    videos:
    - /app/dataset-test-2/panoramic-000-000.mp4
    - /app/dataset-test-2/panoramic-000-001.mp4
    - /app/dataset-test-2/panoramic-000-002.mp4
    - /app/dataset-test-2/panoramic-000-003.mp4
  id: 0
  map: Town03
  pedestrians: 250
  vehicles: 200
  weather: WeatherParameters(cloudyness=80.000000, precipitation=60.000000, precipitation_deposits=100.000000,
    wind_intensity=1.000000, sun_azimuth_angle=0.000000, sun_altitude_angle=15.000000)
- cameras:
  - type: traffic
    videos:
    - /app/dataset-test-2/traffic-004.mp4
  - type: traffic
    videos:
    - /app/dataset-test-2/traffic-005.mp4
  - type: traffic
    videos:
    - /app/dataset-test-2/traffic-006.mp4
  - type: traffic
    videos:
    - /app/dataset-test-2/traffic-007.mp4
  - type: panoramic
    videos:
    - /app/dataset-test-2/panoramic-001-000.mp4
    - /app/dataset-test-2/panoramic-001-001.mp4
    - /app/dataset-test-2/panoramic-001-002.mp4
    - /app/dataset-test-2/panoramic-001-003.mp4
  id: 1
  map: Town07
  pedestrians: 400
  vehicles: 100
  weather: WeatherParameters(cloudyness=80.000000, precipitation=30.000000, precipitation_deposits=50.000000,
    wind_intensity=0.400000, sun_azimuth_angle=0.000000, sun_altitude_angle=75.000000)
timeout: 30
version: 1.0
```

**Comment by BrandonHaynes** [comment]
Hi!  I *think* `town07` might be missing from this build. While I
investigate, if you run again there's a good chance the generator will pick
different maps and you'll survive the process.

Regardless, I'll confirm that my suspicion is correct and release an
update. Thanks for letting me know!

**Comment by yestinchen** [comment]
Hi Brandon, thanks for the quick response! I tried using `scale=2` with other parameters, i.e. different width, height, and seed. It worked sometimes. When it fails, it always fails at `town07`. 
I guess I can live with that for now. Thanks!

By the way, is there any way to change the (average) number of objects?

**Comment by BrandonHaynes** [comment]
Fixed in 9d0f745f70ca2602bade93d7075892f30d969446 (by removing `town07` for now).

I could definitely add switches to override the number of pedestrians and vehicles.  Would that suffice or would you need to fine tune things like bicycles vs motorcycles?  I'll open a separate issue for this feature.

**Comment by yestinchen** [comment]
Yes, that would be enough.   Thanks so much!
